Assets/Scripts/UI/Pickup Pup/DogBrowser.cs

```diff
@@ -0,0 +1,332 @@
+﻿/*
+ * Author(s): Isaiah Mann
+ * Description: Controls the display of a group of dogs in an organized UI
+ * Usage: Should be attached to the parent element in the UI that holds the child elements
+ */
+
+using UnityEngine;
+
+public class DogBrowser : PPUIElement 
```

Comment:
This class doesn't strike me as too long to be acceptable. I think 500+ lines is where a class becomes indisputably too long. I could write a child class for handling all the button logic if you think it's worth revising though. Let me know.